pkg/controller/master/image/vm_image_controller.go

```diff
@@ -142,10 +149,14 @@ func (h *vmImageHandler) checkImage(image *harvesterv1.VirtualMachineImage) (*ha
 
 func (h *vmImageHandler) createBackingImageAndStorageClass(image *harvesterv1.VirtualMachineImage) (*harvesterv1.VirtualMachineImage, error) {
 	if err := h.createBackingImage(image); err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
+		logrus.Errorf("failed to create backing image in image %s/%s, err: %v", image.Namespace, image.Name, err)
+		toUpdate := handleFail(image, condition.Cond(harvesterv1.ImageInitialized), err)
+		return toUpdate, err
 	}
 	if err := h.createStorageClass(image); err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
+		logrus.Errorf("failed to create storage class in image %s/%s, err: %v", image.Namespace, image.Name, err)
+		toUpdate := handleFail(image, condition.Cond(harvesterv1.ImageInitialized), err)
+		return toUpdate, err
```

Comment:
You might need to deep-copy VMImage object first before updating it. Otherwise, you're manipulating the cache, which causes trouble.

Reply:
My bad :sweat:, thanks for the comment. Already address comments in the latest commit. Huge thanks @starbops

Reply:
It would be nice if you could move the line `toUpdate := image.DeepCopy()` to the first line of this function so there won't be multiple `DeepCopy()` calls scattered around.

Reply:
Thanks for the comment ! We can do more as PoAn commented in https://github.com/harvester/harvester/pull/5498#discussion_r1560453534